python/pyarrow/ipc.pxi

```diff
@@ -545,6 +548,9 @@ def read_message(source):
     with nogil:
         check_status(ReadMessage(c_stream, &result.message))
 
+    if result.message == nullptr:
+        raise EOFError("End of Arrow stream")
```

Comment:
You can also have a null message if it was an invalid message (eg `pa.read_message(b"invalidbuffer")` also gives this error now).  Maybe the error message should indicated that? Although this is probably a less typical case to do (I only tried that while trying to find a minimal reproducible example for the segfault).

Reply:
I don't know. We should perhaps make error checking smarter on the C++ side. Can you open a separate JIRA?